### PR TITLE
Add extra `asgi` option with `starlette` as a dependency.

### PR DESCRIPTION
**Pull request by coady**
Follow-up to #1022.

## Description

`asgi` extra installs requirement `starlette`.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #1022

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Follow-up to #1022, by adding `asgi` as an extra.

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Release type: patch
+
+Follow-up to #1022, by adding `asgi` as an extra.
```
Comment (patrick91) [inline RELEASE.md:3]:
````suggestion
Follow-up to [#1022](https://github.com/strawberry-graphql/strawberry/issues/1022), by adding `asgi` as an extra this releases allows to install Strawberry with ASGI support by doing:

```
pip install strawberry[asgi]
```
````

**Review by patrick91 — APPROVED** [review]
Looks good to me, maybe we should update the docs too: https://strawberry.rocks/docs/integrations/asgi

**Comment by BryceBeagle** [comment]
@patrick91 is this ready to merge?

**Review by patrick91 — APPROVED** [review]
Sorry it took me so long to review! Looks good, I've updated the release notes and docs 😊

**Pull request merged**